openstack/networking/v2/extensions/qos/requests.go

```diff
@@ -0,0 +1,9 @@
+package qos
+
+import "github.com/gophercloud/gophercloud"
+
+// ListRuleTypes returns the list of rule types from the server
+func ListRuleTypes(c *gophercloud.ServiceClient) (result ListRuleTypesResult) {
+	_, result.Err = c.Get(listRuleTypesURL(c), &result.Body, nil)
+	return
+}
```

Comment:
With the changes made to `results.go`, this function will now look something like:

```go
// ListRuleTypes retrieves the QoS rule types.
func ListRuleTypes(client *gophercloud.ServiceClient) pagination.Pager {
  url := listRuleTypesURL(c)

  return pagination.NewPager(client, url, func(r pagination.PageResult) pagination.Page {
    return ListRuleTypesPage{pagination.SinglePageBase(r)}
  })
}
```

If you were to put this in a `qos/ruletypes` package, the function name could be changed to just `List`.